app/js/addCategories.ts

```diff
@@ -0,0 +1,45 @@
+async function addCategory ():Promise<void> {
+    let categoriesObject = await getData('/category');
+    document.querySelector<HTMLFormElement>('.categoriesForm').addEventListener('submit', (e) => {
+        e.preventDefault();
+        let newCategory: string = document.querySelector<HTMLFormElement>('#categoryName').value;
+        let responseMessage = document.querySelector('#addedCategoryConfirmation');
+        let listOfCategories = [];
+        categoriesObject.data.forEach((category) => {
+            listOfCategories.push(category.name);
+        })
+
+        if (!listOfCategories.includes(newCategory)) {
+
+            if (newCategory.length > 255) {
+                responseMessage.textContent = 'Error - Category Name Too Long.';
+            } else {
+
+                let data: object = {"name": newCategory};
+                let dataToSend: FormData = jsonToFormData(data);
+                document.querySelector<HTMLFormElement>('#categoryName').value = '';
+
+                sendData(dataToSend, 'category').then((data) => {
+
+                    if (data.success) {
+                        responseMessage.textContent = 'Success! Category added';
+                        responseMessage.classList.remove('alert-danger');
+                        responseMessage.classList.add('alert-success');
+                        location.reload();
```

Comment:
this is not an acceptable solution. you should just be able to run `populateCategories()` here